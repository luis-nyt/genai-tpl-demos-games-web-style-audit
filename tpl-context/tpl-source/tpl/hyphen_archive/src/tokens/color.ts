import { ColorLike, createColorBehaviorStyles, invertColors, setAlpha } from '@nyt/foundation';
import { light as baseLight, dark as baseDark } from '../../../src/tokens';

type ObjectWithContentPrimaryColorLike = { content: { primary: ColorLike } };

const createSemitransparentBg = (alpha: number) => {
  return (clr: ObjectWithContentPrimaryColorLike, name: string) =>
    clr.content.primary.transform(setAlpha(alpha), name);
};

const hoverFilled = createSemitransparentBg(0.16);

const pressedFilled = createSemitransparentBg(0.2);

const hoverOutline = createSemitransparentBg(0.08);

const pressedOutline = createSemitransparentBg(0.12);

/**
 * Extend base colors with hover and outline variants
 */

const light = baseLight.extend.colors({
  hover: {
    filled: hoverFilled(baseDark.colors, 'hover-filled'),
    outline: hoverOutline(baseLight.colors, 'hover-outline'),
    filledDark: hoverFilled(baseLight.colors, 'hover-dark'),
    outlineDark: hoverOutline(baseDark.colors, 'hover-outline-dark'),
  },
  active: {
    filled: pressedFilled(baseDark.colors, 'active-filled'),
    outline: pressedOutline(baseLight.colors, 'active-outline'),
    filledDark: pressedFilled(baseLight.colors, 'active-dark'),
    outlineDark: pressedOutline(baseDark.colors, 'active-outline-dark'),
  },
});

const dark = baseDark.extend.colors({
  hover: {
    filled: hoverFilled(baseLight.colors, 'hover-filled'),
    outline: hoverOutline(baseDark.colors, 'hover-outline'),
  },
  active: {
    filled: pressedFilled(baseLight.colors, 'active-filled'),
    outline: pressedOutline(baseDark.colors, 'active-outline'),
  },
});

const color = light.toVar();

const invertHyphenColors = invertColors(light.colors, dark.colors);

const createHyphenColorBehaviorStyles = createColorBehaviorStyles(light.colors, dark.colors);

export { color, light, dark, invertHyphenColors, createHyphenColorBehaviorStyles };
