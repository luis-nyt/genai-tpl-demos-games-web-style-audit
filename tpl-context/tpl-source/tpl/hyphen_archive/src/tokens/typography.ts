/**
 * fontFamily
 */
const serif = "georgia, 'times new roman', times, serif";
const sans = 'helvetica, arial, sans-serif';
const cheltenham = `'nyt-cheltenham', ${serif}`;
const franklin = `'nyt-franklin', ${sans}`;

export const fontFamily = {
  franklin,
  cheltenham,
};

/**
 * fontWeight
 */
export const fontWeight = {
  200: '200',
  300: '300',
  400: '400',
  600: '600',
  700: '700',
};

/**
 * typeSize
 */
const typeSize = {
  sm: '0.875rem', // 14px
  md: '1rem', // 16px
  lg: '1.125rem', // 18px
  xl: '1.375rem', // 22px
  xxl: '1.625rem', // 26px
};

/**
 * lineHeight
 */
const lineHeight = {
  sm: '0',
  md: '1.3',
  lg: '1.5',
};

/**
 * typography
 */
const typography = {
  core: {
    headline: {
      1: `${fontWeight[600]} ${typeSize.xxl}/${lineHeight.md} ${fontFamily.franklin}`,
      2: `${fontWeight[700]} ${typeSize.xl}/${lineHeight.md} ${fontFamily.franklin}`,
    },
    title: {
      1: `${fontWeight[700]} ${typeSize.lg}/${lineHeight.lg} ${fontFamily.franklin}`,
      2: `${fontWeight[700]} ${typeSize.md}/${lineHeight.lg} ${fontFamily.franklin}`,
      3: `${fontWeight[700]} ${typeSize.sm}/${lineHeight.lg} ${fontFamily.franklin}`,
    },
    subtitle: {
      1: `${fontWeight[600]} ${typeSize.lg}/${lineHeight.lg} ${fontFamily.franklin}`,
      2: `${fontWeight[600]} ${typeSize.md}/${lineHeight.lg} ${fontFamily.franklin}`,
      3: `${fontWeight[600]} ${typeSize.sm}/${lineHeight.lg} ${fontFamily.franklin}`,
    },
    body: {
      1: `${fontWeight[400]} ${typeSize.lg}/${lineHeight.lg} ${fontFamily.franklin}`,
      2: `${fontWeight[400]} ${typeSize.md}/${lineHeight.lg} ${fontFamily.franklin}`,
      3: `${fontWeight[400]} ${typeSize.sm}/${lineHeight.lg} ${fontFamily.franklin}`,
    },
  },
  specialty: {
    display: {
      1: [
        `${fontWeight[300]} 2.5rem/1.2 ${fontFamily.cheltenham}`,
        `${fontWeight[200]} 3.125rem/1.12 ${fontFamily.cheltenham}`,
        `${fontWeight[200]} 2.875rem/1.22 ${fontFamily.cheltenham}`,
      ],
      2: [
        `${fontWeight[300]} 2.25rem/1.2 ${fontFamily.cheltenham}`,
        `${fontWeight[300]} 2.5rem/1.2 ${fontFamily.cheltenham}`,
      ],
      3: `${fontWeight[400]} 1.875rem/1.3333 ${fontFamily.cheltenham}`,
    },
    subscription: {
      1: `${fontWeight[700]} 1.875rem/1.333 ${fontFamily.cheltenham}`,
      2: `${fontWeight[700]} 1.5rem/1.333 ${fontFamily.cheltenham}`,
    },
    legal: {
      1: `${fontWeight[400]} 0.75rem/${lineHeight.lg} ${fontFamily.franklin}`,
    },
  },
};

const typographyFlat = { ...typography.core, ...typography.specialty };

export { typography, typographyFlat };
