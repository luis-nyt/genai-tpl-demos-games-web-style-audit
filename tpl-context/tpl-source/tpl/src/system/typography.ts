import { typography } from '../tokens/index.js';
import { TplSx } from './types.js';

/**
 * Modify an `sx` property to respond to responsive TPL typography values
 */
const tplTypographyModifier = (
  obj: TplSx,
  key: string,
  value,
  breakpoints: string[]
): TplSx | null => {
  if (!['typ', 'typography'].includes(key)) {
    return null;
  }
  /**
   * delete extraneous typ/typography prop
   */
  delete obj[key];

  /**
   * reduce facepaint-style array into target object
   */
  if (Array.isArray(value)) {
    const [init, ...rest] = value;

    return rest.reduce(
      (acc, typName, idx) => {
        const nextBreakpoint = breakpoints[idx];
        if (!typName) {
          return acc;
        }
        acc[nextBreakpoint] = acc[nextBreakpoint] || {};
        acc[nextBreakpoint] = { ...typography[typName], ...acc[nextBreakpoint] };
        return acc;
      },
      { ...typography[init] }
    );
  }
  return { ...typography[value], ...obj };
};

export { tplTypographyModifier };

export type { TplSx };
