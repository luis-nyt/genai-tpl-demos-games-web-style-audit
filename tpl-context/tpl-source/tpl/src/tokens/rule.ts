import { RuleTokens } from '@nyt/foundation';
import { horizontalRuleVariants, verticalRuleVariants } from '../generated/rule.js';

/**
 * @deprecated `ruleTokens` will be replaced in the next major version of `@nyt/tpl`, which will switch to CSS Modules with Sass. Replace with CSS custom properties to achieve similar results.
 * @see {@link https://docs.google.com/document/d/12T5bEefMJDYoLY8gtg0cokrNco53h5t8Ofj8rn5uaV4/edit?pli=1&tab=t.2tb2mnrxg2u6#heading=h.lzd1o6x00ndn TPL Web 2.0 Release Plan: Replace Non-Typography Design Token JavaScript APIs}
 */
const ruleTokens: RuleTokens = {
  horizontal: horizontalRuleVariants,
  vertical: verticalRuleVariants,
};

export { horizontalRuleVariants, verticalRuleVariants, ruleTokens };
