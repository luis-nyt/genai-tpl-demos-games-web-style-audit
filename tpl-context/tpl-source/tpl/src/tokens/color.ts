import { Theme, invertColors, createColorBehaviorStyles } from '@nyt/foundation';
import { lightInput, darkInput } from '../generated/color.js';
import { prefersDarkMode } from '../util/prefersDarkMode.js';

/**
 * @deprecated `light` will be replaced in the next major version of `@nyt/tpl`, which will switch to CSS Modules with Sass. Replace with CSS custom properties to achieve similar results.
 * @see {@link https://docs.google.com/document/d/12T5bEefMJDYoLY8gtg0cokrNco53h5t8Ofj8rn5uaV4/edit?pli=1&tab=t.2tb2mnrxg2u6#heading=h.lzd1o6x00ndn TPL Web 2.0 Release Plan: Replace Non-Typography Design Token JavaScript APIs}
 */
const light = new Theme(lightInput, undefined, 'tpl');

/**
 * @deprecated `dark` will be replaced in the next major version of `@nyt/tpl`, which will switch to CSS Modules with Sass. Replace with CSS custom properties to achieve similar results.
 * @see {@link https://docs.google.com/document/d/12T5bEefMJDYoLY8gtg0cokrNco53h5t8Ofj8rn5uaV4/edit?pli=1&tab=t.2tb2mnrxg2u6#heading=h.lzd1o6x00ndn TPL Web 2.0 Release Plan: Replace Non-Typography Design Token JavaScript APIs}
 */
const dark = new Theme(darkInput, undefined, 'tpl');

const color = light.toVar();

/**
 * @deprecated `invertTplColors` will be removed in the next major version of `@nyt/tpl`, which will switch to CSS Modules with Sass.
 * @see {@link https://docs.google.com/document/d/12T5bEefMJDYoLY8gtg0cokrNco53h5t8Ofj8rn5uaV4/edit?pli=1&tab=t.2tb2mnrxg2u6#heading=h.idkj2loswvck TPL Web 2.0 Release Plan: Remove CSS-in-JS Utilities}
 */
const invertTplColors = invertColors(light.colors, dark.colors);

/**
 * @deprecated `createTplColorBehaviorStyles` will be removed in the next major version of `@nyt/tpl`, which will switch to CSS Modules with Sass. Replace with Color Tokens with Color Behavior, add Color Scheme CSS Classes to parent elements to achieve similar results.
 * @see {@link https://docs.google.com/document/d/12T5bEefMJDYoLY8gtg0cokrNco53h5t8Ofj8rn5uaV4/edit?pli=1&tab=t.2tb2mnrxg2u6#heading=h.278xvmx0xveq TPL Web 2.0 Release Plan: Remove Color Behavior Context & Props}
 */
const createTplColorBehaviorStyles = createColorBehaviorStyles(
  light.colors,
  dark.colors,
  prefersDarkMode
);

export { color, light, dark, invertTplColors, createTplColorBehaviorStyles };
