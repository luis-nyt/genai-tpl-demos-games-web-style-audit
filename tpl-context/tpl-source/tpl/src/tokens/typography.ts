import { fontWeight, fontSize, typography } from '../generated/typography.js';

interface TypeStyle {
  font: string;
  letterSpacing?: `${number}em`;
  textTransform?: 'uppercase';
}

type TypographyKey = keyof typeof typography;

type VariantHelper<Name extends string> =
  Extract<TypographyKey, `${Name}/${string}`> extends `${Name}/${infer U}` ? U : never;

type TypographyPrefix = TypographyKey extends `${infer U}/${string}` ? U : never;

type TypographyVariant<T extends TypographyPrefix> =
  VariantHelper<T> extends infer U
    ? /** @see https://stackoverflow.com/a/69090186 */
      U extends `${infer Digit extends number}`
      ? Digit
      : U
    : never;

export { fontSize, fontWeight, typography };
export type { TypographyPrefix, TypographyVariant, TypeStyle, TypographyKey };
