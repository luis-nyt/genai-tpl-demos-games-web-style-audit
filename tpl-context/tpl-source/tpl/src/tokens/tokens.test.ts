import { Theme } from '@nyt/foundation';
import { lightInput, darkInput } from '../generated/color.js';
import { fontWeight, fontSize, typography } from '../generated/typography.js';
import { borderWidth } from '../generated/borderWidth.js';
import { spaceScale } from '../generated/spacing.js';
import { ruleTokens } from './rule.js';
import cssVars from '../generated/cssVar.js';

const snaps: [string, any][] = [
  ['lightInput', lightInput],
  ['darkInput', darkInput],
  ['fontSize', fontSize],
  ['typography', typography],
  ['fontWeight', fontWeight],
  ['borderWidth', borderWidth],
  ['ruleTokens', ruleTokens],
  ['spaceScale', spaceScale],
  ['cssVars', cssVars],
];

describe('generated tokens', () => {
  test.each(snaps)('%s', (_testName, tokenObj) => {
    expect(tokenObj).toMatchSnapshot();
  });

  test('instance of Theme using lightInput can be created without errors', () => {
    expect(() => new Theme(lightInput)).not.toThrow();
  });

  test('instance of Theme using darkInput can be created without errors', () => {
    expect(() => new Theme(darkInput)).not.toThrow();
  });
});
