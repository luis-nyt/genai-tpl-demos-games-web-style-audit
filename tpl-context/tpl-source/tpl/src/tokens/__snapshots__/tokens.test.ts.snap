// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generated tokens borderWidth 1`] = `
Map {
  1 => "0.0625rem",
  2 => "0.125rem",
  1.5 => "0.09375rem",
}
`;

exports[`generated tokens cssVars 1`] = `
Object {
  "--cheltenham": "\\"nyt-cheltenham\\",var(--serif)",
  "--cheltenham-condensed": "\\"nyt-cheltenham-cond\\",var(--serif)",
  "--cheltenham-text-condensed": "\\"nyt-cheltenham-text-cond\\",var(--serif)",
  "--franklin": "\\"nyt-franklin\\",var(--sans-serif)",
  "--imperial": "\\"nyt-imperial\\",var(--serif)",
  "--karnak": "\\"nyt-karnak\\",var(--serif)",
  "--sans-serif": "helvetica, arial, sans-serif",
  "--serif": "georgia, 'times new roman', times, serif",
}
`;

exports[`generated tokens darkInput 1`] = `
Object {
  "background": Object {
    "highlight": "#35352F",
    "primary": "#121212",
    "scrim": Array [
      0,
      0,
      7,
      0.6,
    ],
    "secondary": "#2A2A2A",
    "tertiary": "#424242",
  },
  "content": Object {
    "accent": "#6BA1DD",
    "accentDim": "#7CA7C8",
    "breaking": "#E14E5B",
    "negative": "#EA7980",
    "placeholder": "#8B8B8B",
    "positive": "#63A859",
    "primary": "#F8F8F8",
    "primaryDim": "#DFDFDF",
    "secondary": "#BBBBBB",
    "secondaryDim": "#A3A3A3",
  },
  "stroke": Object {
    "primary": "#F8F8F8",
    "secondary": "#979797",
    "tertiary": "#424242",
  },
}
`;

exports[`generated tokens fontSize 1`] = `
Array [
  "0.6875rem",
  "0.75rem",
  "0.875rem",
  "1rem",
  "1.125rem",
  "1.25rem",
  "1.375rem",
  "1.5rem",
  "1.75rem",
  "2rem",
  "2.25rem",
  "2.5rem",
  "3rem",
  "3.5rem",
  "4rem",
  "4.5rem",
]
`;

exports[`generated tokens fontWeight 1`] = `
Object {
  "black": 900,
  "bold": 700,
  "book": 400,
  "extraBold": 800,
  "extraLight": 200,
  "light": 300,
  "medium": 500,
  "normal": 400,
  "regular": 400,
  "semiBold": 600,
}
`;

exports[`generated tokens lightInput 1`] = `
Object {
  "background": Object {
    "highlight": "#FEFAD1",
    "primary": "#FFFFFF",
    "scrim": Array [
      0,
      0,
      7,
      0.6,
    ],
    "secondary": "#F8F8F8",
    "tertiary": "#EBEBEB",
  },
  "content": Object {
    "accent": "#346EB7",
    "accentDim": "#326891",
    "breaking": "#D0021B",
    "negative": "#A90111",
    "placeholder": "#8B8B8B",
    "positive": "#267C30",
    "primary": "#121212",
    "primaryDim": "#363636",
    "secondary": "#5A5A5A",
    "secondaryDim": "#727272",
  },
  "stroke": Object {
    "primary": "#121212",
    "secondary": "#979797",
    "tertiary": "#DFDFDF",
  },
}
`;

exports[`generated tokens ruleTokens 1`] = `
Object {
  "horizontal": Object {
    "primary": "0.125rem solid var(--color-stroke-primary, #121212)",
    "secondary": "0.0625rem solid var(--color-stroke-primary, #121212)",
    "tertiary": "0.0625rem solid var(--color-stroke-tertiary, #DFDFDF)",
  },
  "vertical": Object {
    "primary": "0.0625rem solid var(--color-stroke-primary, #121212)",
    "secondary": "0.0625rem solid var(--color-stroke-tertiary, #DFDFDF)",
  },
}
`;

exports[`generated tokens spaceScale 1`] = `
Map {
  1 => "0.5rem",
  2 => "1rem",
  3 => "1.5rem",
  4 => "2rem",
  5 => "2.5rem",
  6 => "3rem",
  7 => "3.5rem",
  8 => "4rem",
  9 => "4.5rem",
  10 => "5rem",
  0.5 => "0.25rem",
  1.5 => "0.75rem",
  2.5 => "1.25rem",
}
`;

exports[`generated tokens typography 1`] = `
Object {
  "Body/14": Object {
    "font": "400 0.875rem/1.39 nyt-imperial, georgia, 'times new roman', times, serif",
  },
  "Body/16": Object {
    "font": "400 1rem/1.39 nyt-imperial, georgia, 'times new roman', times, serif",
  },
  "Body/18": Object {
    "font": "400 1.125rem/1.39 nyt-imperial, georgia, 'times new roman', times, serif",
  },
  "Body/20": Object {
    "font": "400 1.25rem/1.5 nyt-imperial, georgia, 'times new roman', times, serif",
  },
  "Headline/16": Object {
    "font": "500 1rem/1.2 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.01em",
  },
  "Headline/18": Object {
    "font": "400 1.125rem/1.2 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.01em",
  },
  "Headline/24": Object {
    "font": "400 1.5rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "Headline/28": Object {
    "font": "400 1.75rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineFeature/24": Object {
    "font": "300 1.5rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineFeature/28": Object {
    "font": "200 1.75rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineFeature/32": Object {
    "font": "200 2rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineFeature/48": Object {
    "font": "200 3rem/1.1 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineFeature/56": Object {
    "font": "200 3.5rem/1 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineNews/16": Object {
    "font": "700 1rem/1.2 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.01em",
  },
  "HeadlineNews/18": Object {
    "font": "700 1.125rem/1.2 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.01em",
  },
  "HeadlineNews/20": Object {
    "font": "700 1.25rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.01em",
  },
  "HeadlineNews/22": Object {
    "font": "700 1.375rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineNews/28": Object {
    "font": "700 1.75rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineNews/32": Object {
    "font": "700 2rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineNews/36": Object {
    "font": "700 2.25rem/1.15 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineNews/40": Object {
    "font": "700 2.5rem/1.1 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineNews/48": Object {
    "font": "700 3rem/1.1 nyt-cheltenham, cheltenham-fallback-georgia, cheltenham-fallback-noto, georgia, 'times new roman', times, serif",
  },
  "HeadlineOpinion/18": Object {
    "font": "700 1.125rem/1.15 nyt-cheltenham-text-cond, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.0025em",
  },
  "HeadlineOpinion/20": Object {
    "font": "700 1.25rem/1.15 nyt-cheltenham-text-cond, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.0025em",
  },
  "HeadlineOpinion/22": Object {
    "font": "700 1.375rem/1.15 nyt-cheltenham-text-cond, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.0025em",
  },
  "HeadlineOpinion/24": Object {
    "font": "700 1.5rem/1.15 nyt-cheltenham-text-cond, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.0025em",
  },
  "HeadlineOpinion/36": Object {
    "font": "700 2.25rem/1.15 nyt-cheltenham-text-cond, georgia, 'times new roman', times, serif",
    "letterSpacing": "0.0025em",
  },
  "HeadlineOpinion/56": Object {
    "font": "700 3.5rem/1.1 nyt-cheltenham-cond, georgia, 'times new roman', times, serif",
  },
  "Label/Emphasis": Object {
    "font": "800 0.6875rem/1.25 nyt-franklin, helvetica, arial, sans-serif",
    "letterSpacing": "0.1em",
    "textTransform": "uppercase",
  },
  "Label/Regular": Object {
    "font": "500 0.6875rem/1.25 nyt-franklin, helvetica, arial, sans-serif",
    "letterSpacing": "0.1em",
    "textTransform": "uppercase",
  },
  "Text/12": Object {
    "font": "500 0.75rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Text/14": Object {
    "font": "500 0.875rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Text/16": Object {
    "font": "500 1rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Text/18": Object {
    "font": "500 1.125rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Text/20": Object {
    "font": "500 1.25rem/1.2 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/12": Object {
    "font": "600 0.75rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/14": Object {
    "font": "600 0.875rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/16": Object {
    "font": "600 1rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/18": Object {
    "font": "600 1.125rem/1.3 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/20": Object {
    "font": "600 1.25rem/1.2 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/22": Object {
    "font": "600 1.375rem/1.2 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/24": Object {
    "font": "600 1.5rem/1.2 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/28": Object {
    "font": "600 1.75rem/1.2 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/32": Object {
    "font": "600 2rem/1.2 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/48": Object {
    "font": "600 3rem/1.1 nyt-franklin, helvetica, arial, sans-serif",
  },
  "Title/56": Object {
    "font": "600 3.5rem/1 nyt-franklin, helvetica, arial, sans-serif",
  },
  "TitleKarnak/16": Object {
    "font": "700 1rem/1.15 nyt-karnak, georgia, 'times new roman', times, serif",
  },
  "TitleKarnak/18": Object {
    "font": "700 1.125rem/1.15 nyt-karnak, georgia, 'times new roman', times, serif",
  },
}
`;
